src/formatDate.js

```diff
@@ -8,7 +8,42 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separatorFrom = fromFormat.pop();
+  const separatorTo = toFormat.pop();
+  const dateParts = date.split(separatorFrom);
+  const newDate = {};
+
+  for (let i = 0; i < dateParts.length; i++) {
+    const key = fromFormat[i];
+    const value = dateParts[i];
+
+    newDate[key] = value;
+  }
+
+  const resultDate = toFormat
+    .map((e) => {
+      if (!newDate[e]) {
+        if (e === 'YY') {
+          return newDate.YYYY.slice(-2);
+        } else {
+          const end = newDate.YY;
+          let start;
+
+          if (end >= 30) {
+            start = 19;
+          } else {
+            start = 20;
+          }
```

Comment:
The logic for determining the century is correct, but it's better to encapsulate this logic in a separate function for clarity and potential reusability.